packages/api-client-core/src/InternalModelManager.ts

```diff
@@ -203,7 +203,7 @@ export class InternalModelManager {
 
   async maybeFindOne(id: string): Promise<GadgetRecord<any> | null> {
     const record = await this.findOne(id, false);
-    return record ?? null;
+    return record && "id" in record ? record : null;
```

Comment:
Interesting,`this.findOne(id, false)` can return a non-null object without an id?

Should that even be allowed to happen?

Reply:
So I don't think this could really happen previously (not sure about the hook, but the func couldn't), because it would throw if the data was empty.

BUT... once I added the ability to not throw on empty data (for the `maybeFindOne` func), it meant that the code could execute past that step, onto the hydration step, which still resulted in an object being created with a few basic keys. So I'm checking for the one key that it will always have if the data is not empty.

Reply:
So I missed your point, which @thegedge reiterated as well — given the output type of `this.findOne`, a `null` output should not be possible.

Based on his suggestion, I've tightened up the typing around `GadgetRecord` — similar to what you had done with `unknown`, but a bit more specific, with an `extends` based on `Record<string, any>`.